File: src/main/java/org/kite9/diagram/visualization/planarization/mgt/MGTPlanarizer.java
package org.kite9.diagram.visualization.planarization.mgt;

import org.kite9.diagram.adl.Diagram;
import org.kite9.diagram.common.elements.mapping.ElementMapper;
import org.kite9.diagram.visualization.planarization.AbstractPlanarizer;
import org.kite9.diagram.visualization.planarization.Planarization;
import org.kite9.diagram.visualization.planarization.PlanarizationBuilder;
import org.kite9.diagram.visualization.planarization.Planarizer;
import org.kite9.diagram.visualization.planarization.mgt.builder.HierarchicalPlanarizationBuilder;
import org.kite9.diagram.visualization.planarization.mgt.face.FaceConstructor;
import org.kite9.diagram.visualization.planarization.mgt.face.FaceConstructorImpl;
import org.kite9.diagram.visualization.planarization.rhd.position.PositionRoutableHandler2D;
import org.kite9.diagram.visualization.planarization.rhd.position.RoutableHandler2D;

public class MGTPlanarizer extends AbstractPlanarizer implements Planarizer {

	public MGTPlanarizer(ElementMapper elementMapper) {
		super(elementMapper);
	}

	protected FaceConstructor getFaceConstructor() {
		return new FaceConstructorImpl();
	}
	
	protected RoutableHandler2D  getRoutableHandler() {
		return new PositionRoutableHandler2D();
	}

	@Override
	protected PlanarizationBuilder getPlanarizationBuilder() {
		return new HierarchicalPlanarizationBuilder(getElementMapper(), getGridPositioner());
	}

	@Override
	protected Planarization buildPlanarization(Diagram c) {
		Planarization pln = super.buildPlanarization(c);
		getFaceConstructor().createFaces((MGTPlanarization) pln);
		return pln;
	}
	
	

}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/mgt/MGTPlanarizer.kt
package org.kite9.diagram.visualization.planarization.mgt


import org.kite9.diagram.common.elements.mapping.ElementMapper
import org.kite9.diagram.model.Diagram
import org.kite9.diagram.visualization.planarization.AbstractPlanarizer
import org.kite9.diagram.visualization.planarization.Planarization
import org.kite9.diagram.visualization.planarization.PlanarizationBuilder
import org.kite9.diagram.visualization.planarization.mgt.builder.HierarchicalPlanarizationBuilder
import org.kite9.diagram.visualization.planarization.mgt.face.FaceConstructor
import org.kite9.diagram.visualization.planarization.mgt.face.FaceConstructorImpl
import org.kite9.diagram.visualization.planarization.rhd.position.PositionRoutableHandler2D
import org.kite9.diagram.visualization.planarization.rhd.position.RoutableHandler2D

class MGTPlanarizer(elementMapper: ElementMapper) : AbstractPlanarizer(elementMapper) {

    override val faceConstructor: FaceConstructor = FaceConstructorImpl()

    protected val routableHandler: RoutableHandler2D = PositionRoutableHandler2D()
    override val planarizationBuilder: PlanarizationBuilder = HierarchicalPlanarizationBuilder(elementMapper, gridPositioner)

    override fun buildPlanarization(c: Diagram): Planarization {
        val pln = super.buildPlanarization(c)
        faceConstructor.createFaces((pln as MGTPlanarization))
        return pln
    }
}